### ISSUE TITLE
Improve the error when imports are written before a `part of` directive

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#57044]
With enhanced parts, it seems `import`s should be before `part` directives, but they come after `part of` directives. This makes some sense, but it's easy to get wrong. If I put an import after a `part`, I get a clear error message:

![image](https://github.com/user-attachments/assets/0ac8197e-7482-4866-94fc-3934205ce701)

However the same is not true if I get a `part of` and import the wrong way around:

![image](https://github.com/user-attachments/assets/43bc92b9-a8a3-469e-9896-962ccf4bcfea)

The error makes it seem like I can't have imports in this file, but in reality I just need to put them after the `part of`.

**Comment by bwilkerson** [comment]
Agreed. There's a lot of work that needs to be done before we can ship the enhanced-parts feature, but we aren't actively working on that feature at the moment.
